projects/flimnap/main.cc

```diff
@@ -65,10 +65,30 @@ namespace {
     constexpr float fixed_velocity = 5.0f;
 
     // pose calculation loop
-    constexpr systime_t pose_loop_time = US2ST(8333); // update pose at 120 Hz
+    constexpr systime_t pose_loop_period = US2ST(8333); // update pose at 120 Hz
 
     // dynamics loop
-    constexpr systime_t looptime = MS2ST(1); // 1 ms -> 1 kHz
+    constexpr systime_t dynamics_loop_period = MS2ST(1); // 1 ms -> 1 kHz
+
+    // Suspends the invoking thread until the system time arrives to the
+    // specified value.
+    // The system time is assumed to be between prev and time else the call is
+    // assumed to have been called outside the allowed time interval, in this
+    // case the thread yields the time slot and no sleep is performed.
+    systime_t chThdSleepUntilWindowedOrYield(systime_t prev, systime_t next) {
+        systime_t time;
+
+        chSysLock();
+        time = chVTGetSystemTimeX();
+        if (chVTIsTimeWithinX(time, prev, next)) {
+            chThdSleepS(next - time);
+        } else {
+            chSchDoYieldS();
+        }
+        chSysUnlock();
```

Comment:
I assume the `chSysLock()` and `chSysUnlock()` are used so that in between the query of `time` and the sleep, no other work is being done causing a too long sleep? Is the increased accuracy worth the time spent locking and unlocking?